packages/react-vapor/src/components/flatSelect/FlatSelectSelectors.ts

```diff
@@ -0,0 +1,16 @@
+import {createSelector} from 'reselect';
+import * as _ from 'underscore';
+
+import {IReactVaporState} from '../../ReactVapor';
+import {flatSelectInitialState, IFlatSelectState } from './FlatSelectReducers';
+
+const getInput = (state: IReactVaporState, props: {id: string}): IFlatSelectState => {
+    return _.findWhere(state.flatSelect, {id: props.id}) || flatSelectInitialState;

```

Comment:
I think it would be best to let it return `undefined` instead of the flatSelectInitialState. From a conceptual standpoint, the flatSelect is really undefined in the state, so why should we lie about it :wink:

Reply:
`        it('should return the default flatSelect state when the flatSelect is not in the store', () => {
            const FlatSelect = FlatSelectSelectors.getInput({ flatSelect: [] }, { id: 'nothing is selected by the user' });
            expect(FlatSelect).toEqual(flatSelectInitialState);
        });`

Reply:
didn't you write the test?

Reply:
YES but my point is to expect that if nothing is selected by the user, the initial value is chosen.
but... I guess it is useless right ?